### ISSUE TITLE
Cron schedule bug

### ISSUE THREAD

**Issue by adrok** [magento/magento2#4173]
## Steps to reproduce
1. Install Magento from `develop` branch.
2. run bin/magento cron:run
## Expected result
1. With default configuration, generated 4 jobs for "newsletter_send_all" (every 5 minutes, ahead 20 minutes)
## Actual result
1. Generated one job

---

magento2/app/code/Magento/Cron/Observer/ProcessCronQueueObserver.php wrong return statement (for cycle)

```
if ($schedule->trySchedule()) {
                // time matches cron expression
                $schedule->save();
                return; //remove it!
            }

```

**Comment by mazhalai** [comment]
@adrok thank you for reporting, we have MAGETWO-51939 to investigate and fix.

**Comment by kanduvisla** [comment]
Could this also be related with an issue I am having at the moment? I'm trying to schedule multiple jobs in one group, with the same instance, but different methods:

```
<job name="sync_inventory_5m" instance="Vendor\Module\Cron\SynchronizeInventory" method="execute_5m">
    <schedule>*/5 * * * *</schedule>
</job>
<job name="sync_inventory_10m" instance="Vendor\Module\Cron\SynchronizeInventory" method="execute_10m">
    <schedule>*/10 * * * *</schedule>
</job>
<job name="sync_inventory_15m" instance="Vendor\Module\Cron\SynchronizeInventory" method="execute_15m">
    <schedule>*/15 * * * *</schedule>
</job>
<job name="sync_inventory_30m" instance="Vendor\Module\Cron\SynchronizeInventory" method="execute_30m">
    <schedule>*/30 * * * *</schedule>
</job>
<job name="sync_inventory_1h" instance="Vendor\Module\Cron\SynchronizeInventory" method="execute_1h">
    <schedule>0 * * * *</schedule>
</job>
<job name="sync_inventory_2h" instance="Vendor\Module\Cron\SynchronizeInventory" method="execute_2h">
    <schedule>0 */2 * * *</schedule>
</job>
<job name="sync_inventory_3h" instance="Vendor\Module\Cron\SynchronizeInventory" method="execute_3h">
    <schedule>0 */3 * * *</schedule>
</job>
<job name="sync_inventory_6h" instance="Vendor\Module\Cron\SynchronizeInventory" method="execute_6h">
    <schedule>0 */6 * * *</schedule>
</job>
<job name="sync_inventory_12h" instance="Vendor\Module\Cron\SynchronizeInventory" method="execute_12h">
    <schedule>0 */12 * * *</schedule>
</job>
<job name="sync_inventory_24h" instance="Vendor\Module\Cron\SynchronizeInventory" method="execute_24h">
    <schedule>0 0 * * *</schedule>
</job>
```

The result in the database table are crons for 5m, 10m and 15m and sometimes 30m. Could it be that Magento 2 thinks these are duplicate crons and thinks _'hey 5 minutes also fits in 10 and 15, so I don't need te re-schedule it'._ Fact is that I do need to run those different methods on different intervals

**Comment by kanduvisla** [comment]
Perhaps related to this bug: tested in M2.1.1: when I add a cronjob that needs to be executed every 5 minutes, it's executed every 20 minutes or so.

**Comment by adrian-martinez-interactiv4** [comment]
Why isn't this fix already applied at 2.1.7 version ? :(

**Comment by korostii** [comment]
@magento-team, please reopen this until the fix is released in a 2.1.x version.

**Comment by magento-engcom-team** [comment]
Hi @adrok. Thank you for your report. 
The issue has been fixed in magento/magento2#14096 by @simpleadm in 2.1-develop branch
Related commit(s):
 - [1a8e4a56988238ea3947ea61337d3987fd9af10d](https://github.com/magento/magento2/commit/1a8e4a56988238ea3947ea61337d3987fd9af10d)

The fix will be available with the upcoming 2.1.14 release.
